hotel/models/hotel_folio.py

```diff
@@ -0,0 +1,753 @@
+
+import time
+from datetime import datetime, timedelta
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT
+
+
+def _offset_format_timestamp1(
+    src_tstamp_str,
+    src_format,
+    dst_format,
+    ignore_unparsable_time=True,
+    context=None,
+):
+    """
+    Convert a source timeStamp string into a destination timeStamp string,
+    attempting to apply the correct offset if both the server and local
+    timeZone are recognized,or no offset at all if they aren't or if
+    tz_offset is false (i.e. assuming they are both in the same TZ).
+    @param src_tstamp_str: the STR value containing the timeStamp.
+    @param src_format: the format to use when parsing the local timeStamp.
+    @param dst_format: the format to use when formatting the resulting
+     timeStamp.
+    @param server_to_client: specify timeZone offset direction (server=src
+                             and client=dest if True, or client=src and
+                             server=dest if False)
+    @param ignore_unparsable_time: if True, return False if src_tstamp_str
+                                   cannot be parsed using src_format or
+                                   formatted using dst_format.
+    @return: destination formatted timestamp, expressed in the destination
+             timezone if possible and if tz_offset is true, or src_tstamp_str
+             if timezone offset could not be determined.
+    """
+    if not src_tstamp_str:
+        return False
+    res = src_tstamp_str
+    if src_format and dst_format:
+        try:
+            # dt_value needs to be a datetime object\
+            # (so notime.struct_time or mx.DateTime.DateTime here!)
+            dt_value = datetime.strptime(src_tstamp_str, src_format)
+            if context.get("tz", False):
+                try:
+                    import pytz
+
+                    src_tz = pytz.timezone(context["tz"])
+                    dst_tz = pytz.timezone("UTC")
+                    src_dt = src_tz.localize(dt_value, is_dst=True)
+                    dt_value = src_dt.astimezone(dst_tz)
+                except Exception:
+                    pass
+            res = dt_value.strftime(dst_format)
+        except Exception:
+            # Normal ways to end up here are if strptime or strftime failed
+            if not ignore_unparsable_time:
+                return False
+    return res
+
+
+class FolioRoomLine(models.Model):
+
+    _name = "folio.room.line"
+    _description = "Hotel Room Reservation"
+    _rec_name = "room_id"
+
+    room_id = fields.Many2one("hotel.room", "Room id")
+    check_in = fields.Datetime("Check In Date", required=True)
+    check_out = fields.Datetime("Check Out Date", required=True)
+    folio_id = fields.Many2one("hotel.folio", "Folio Number")
+    status = fields.Selection(string="state", related="folio_id.state")
```

Comment:
```suggestion
    status = fields.Selection(string="State", related="folio_id.state")
```
or It should be Status